internal/build/phases_test.go

```diff
@@ -0,0 +1,505 @@
+				configProvider.ContainerConfig().Cmd,
+				[]string{"-log-level", "debug"},
+				[]string{"-cache-dir", "/cache"},
+				[]string{"-layers", "/layers"},
+			)
+		})
+
+		it("configures the phase with binds", func() {
+			lifecycle := fakeLifecycle(t, false)
+			fakePhaseFactory := fakes.NewFakePhaseFactory()
+			expectedBinds := []string{"some-cache:/cache"}
+
+			err := lifecycle.Restore(context.Background(), "some-cache", fakePhaseFactory)
+			h.AssertNil(t, err)
+
+			configProvider := fakePhaseFactory.NewCalledWithProvider
+			h.AssertIncludeAllExpectedPatterns(t, configProvider.HostConfig().Binds, expectedBinds)
+		})
+	})
+
+	when("#Analyze", func() {
+		it("creates a phase and then runs it", func() {
+			lifecycle := fakeLifecycle(t, false)
+			fakePhase := &fakes.FakePhase{}
+			fakePhaseFactory := fakes.NewFakePhaseFactory(fakes.WhichReturnsForNew(fakePhase))
+
+			err := lifecycle.Analyze(context.Background(), "test", "test", false, false, fakePhaseFactory)
+			h.AssertNil(t, err)
+
+			h.AssertEq(t, fakePhase.CleanupCallCount, 1)
+			h.AssertEq(t, fakePhase.RunCallCount, 1)
+		})
+
+		when("clear cache", func() {
+			it("configures the phase with the expected arguments", func() {
+				lifecycle := fakeLifecycle(t, false)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+				expectedRepoName := "some-repo-name"
+
+				err := lifecycle.Analyze(context.Background(), expectedRepoName, "test", false, true, fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				h.AssertEq(t, fakePhaseFactory.NewCalledWithName, "analyzer")
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertIncludeAllExpectedPatterns(t,
+					configProvider.ContainerConfig().Cmd,
+					[]string{"-skip-layers"},
+				)
+			})
+		})
+
+		when("clear cache is false", func() {
+			it("configures the phase with the expected arguments", func() {
+				lifecycle := fakeLifecycle(t, false)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+				expectedRepoName := "some-repo-name"
+
+				err := lifecycle.Analyze(context.Background(), expectedRepoName, "test", false, false, fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				h.AssertEq(t, fakePhaseFactory.NewCalledWithName, "analyzer")
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertIncludeAllExpectedPatterns(t,
+					configProvider.ContainerConfig().Cmd,
+					[]string{"-cache-dir", "/cache"},
+				)
+			})
+		})
+
+		when("publish", func() {
+			it("configures the phase with registry access", func() {
+				lifecycle := fakeLifecycle(t, false)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+				expectedRepos := []string{"some-repo-name"}
+
+				err := lifecycle.Analyze(context.Background(), expectedRepos[0], "test", true, false, fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				configProvider := fakePhaseFactory.NewCalledWithProvider
+				h.AssertIncludeAllExpectedPatterns(t, configProvider.ContainerConfig().Env, []string{"CNB_REGISTRY_AUTH={}"})
```

Comment:
Is this expectation okay? That is indeed the value that is set in the environment, but I am surprised it does not reference the repo name that was passed in.